### PR TITLE
Improved downloading experience

### PR DESCRIPTION
**Pull request by rishab247**
#### What is it?
- [ ] Bugfix (user facing)
- [X] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
- Added a new Dialog box which will be displayed between when a download is initiated (by sharing a youtube video with the newPipe App) and DownloadDialog is shown.
image

#### Before/After Screenshots/Screen Record

- Before:
[Screen_recording_20230910_235412.webm](https://github.com/TeamNewPipe/NewPipe/assets/47221639/8bd6df32-5b1f-4c07-93c6-5c6bd1d64f76)

- After:
[Screen_recording_20230910_234930.webm](https://github.com/TeamNewPipe/NewPipe/assets/47221639/fad0c1f7-d862-4d41-8506-63cffaa62a00)

#### Fixes the following issue(s)

- Fixes https://github.com/TeamNewPipe/NewPipe/issues/10378
Preventing user from dropping out of the download flow while waiting for DownloadDialog to appear.

#### APK testing

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR. You can find more info and a video demonstration [on this wiki page](https://github.com/TeamNewPipe/NewPipe/wiki/Download-APK-for-PR).

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/download/LoadingDialog.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    @Override
+    public void onDismiss(@NonNull final DialogInterface dialog) {
+        super.onDismiss(dialog);
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+    }
```
Comment (TobiGr) [inline app/src/main/java/org/schabi/newpipe/download/LoadingDialog.java:78]: Why are those methods overridden?
This code can be removed.

File: app/src/main/java/org/schabi/newpipe/download/LoadingDialog.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    @Override
+    public void onDestroyView() {
+        dialogLoadingBinding = null;
+        super.onDestroyView();
+    }
+
+    @Override
+    public void onSaveInstanceState(@NonNull final Bundle outState) {
+        super.onSaveInstanceState(outState);
+    }
```
Comment (TobiGr) [inline app/src/main/java/org/schabi/newpipe/download/LoadingDialog.java:89]: Why is this method overridden?
The code can be removed.

File: app/src/main/java/org/schabi/newpipe/download/LoadingDialog.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        super.onCreate(savedInstanceState);
+        if (DEBUG) {
+            Log.d(TAG, "onCreate() called with: "
+                    + "savedInstanceState = [" + savedInstanceState + "]");
+        }
+        if (!PermissionHelper.checkStoragePermissions(getActivity(),
+                PermissionHelper.DOWNLOAD_DIALOG_REQUEST_CODE)) {
+            dismiss();
+            return;
+        }
```
Comment (TobiGr) [inline app/src/main/java/org/schabi/newpipe/download/LoadingDialog.java:37]: I don't understand why the dialog is dismissed when permissions are missing. That's already handled in the DownloadDialog.

File: app/src/main/java/org/schabi/newpipe/download/LoadingDialog.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        super.onViewCreated(view, savedInstanceState);
+        dialogLoadingBinding = DownloadLoadingDialogBinding.bind(view);
+        initToolbar(dialogLoadingBinding.toolbarLayout.toolbar);
+    }
+
+    private void initToolbar(final Toolbar toolbar) {
+        if (DEBUG) {
+            Log.d(TAG, "initToolbar() called with: toolbar = [" + toolbar + "]");
+        }
+        toolbar.setTitle("Loading Metadata...");
```
Comment (TobiGr) [inline app/src/main/java/org/schabi/newpipe/download/LoadingDialog.java:65]: String res not used.

**Review by TobiGr — CHANGES_REQUESTED** [review]
Thank you for the work.

**Comment by TobiGr** [comment]
I applied some of my review comments and made the dialog's title customizable to allow using it in other part's of the app, too.

File: app/src/main/java/org/schabi/newpipe/download/LoadingDialog.java (hunk 3 above)
Comment (rishab247) [reply app/src/main/java/org/schabi/newpipe/download/LoadingDialog.java:55]: missed it! removing...

**Review by rishab247 — COMMENTED** [review]
@TobiGr i am getting this error from CI `Unknown failure: cmd: Failure calling service package: Broken pipe (32)` this looks like a pipline issue as `:app:connectedDebugAndroidTest` working fine on my local. can you please rerun the pipeline

**Review by TobiGr — APPROVED** [review]
LGTM

(please squash when merging)

**Pull request merged**